offline/cable_cru_TRENDY.F90

```diff
@@ -1,1318 +1,1069 @@
+  ! derived type.
+
+  ! Internally, we're going to pick up the individually separated inputs and
+  ! pack them into a more convenient data structure. We don't expect the user
+  ! to know the order in which we store the MET inputs internally, so we need
+  ! to access them by name.
+  CHARACTER(LEN=256), DIMENSION(13), INTENT(OUT) :: InputFiles
+
+  TYPE(CRU_TYPE), INTENT(OUT)  :: CRU    ! The master CRU structure
+
+  ! Now we assign variables to temporarily store the files in the namelist
+  ! In an ideal world, we'd have the users write directory to the InputFiles
+  ! string array, but we can't expect everyone to know the order the files are
+  ! expected in. So we first read them from a recognisable name, then pass them
+  ! to the array.
+  CHARACTER(LEN=256)  :: rainFile, lwdnFile, swdnFile, presFile, qairFile,&
+                         TmaxFile, TminFile, uwindFile, vwindFile, fdiffFile,&
+                         CO2File, NDepFile, landmaskFile
+  LOGICAL             :: rainRecycle, lwdnRecycle, swdnRecycle, presRecycle,&
+                         qairRecycle, TmaxRecycle, TminRecycle, uwindRecycle,&
+                         vwindRecycle, fdiffRecycle
+  CHARACTER(LEN=16)   :: CO2Method, NDepMethod
+  INTEGER             :: MetRecyc, RecycStartYear
+  REAL                :: DtHrs
+  LOGICAL             :: ReadDiffFrac, LeapYears, DirectRead
+
+  ! Need a unit to handle the io
+  INTEGER             :: nmlUnit
+
+  ! Set the initial values for the filenames, as there are many instances where
+  ! not all are required. We will set them all initially to "None", which we
+  ! can easily check against later.
+  rainFile = "None"
+  lwdnFile = "None"
+  swdnFile = "None"
+  presFile = "None"
+  qairFile = "None"
+  TmaxFile = "None"
+  TminFile = "None"
+  uwindFile = "None"
+  vwindFile = "None"
+  fdiffFile = "None"
+  CO2File = "None"
+  NDepFile = "None"
+  landmaskFile = "None"
+
+  ! For the recycling booleans
+  rainRecycle = .FALSE.
+  lwdnRecycle = .FALSE.
+  swdnRecycle = .FALSE.
+  presRecycle = .FALSE.
+  qairRecycle = .FALSE.
+  TmaxRecycle = .FALSE.
+  TminRecycle = .FALSE.
+  uwindRecycle = .FALSE.
+  vwindRecycle = .FALSE.
+  fdiffRecycle = .FALSE.
+
+  ! Defaults for the other inputs
+  CO2Method = "Yearly"
+  NDepMethod = "Yearly"
+  MetRecyc = 20
+  RecycStartYear = 1901
+  DtHrs = 3.0
+  LeapYears = .FALSE.
+  ReadDiffFrac = .TRUE.
+  DirectRead = .FALSE.
+
+  ! Set up and read the namelist
+  NAMELIST /crunml/ rainFile, lwdnFile, swdnFile, presFile, qairFile,&
+                    TmaxFile, TminFile, uwindFile, vwindFile, fdiffFile,&
+                    CO2File, NDepFile, landmaskFile,&
+                    rainRecycle, lwdnRecycle, swdnRecycle, presRecycle,&
+                    qairRecycle, TmaxRecycle, TminRecycle, uwindRecycle,&
+                    vwindRecycle, fdiffRecycle,&
+                    ReadDiffFrac, CO2Method, NDepMethod, MetRecyc, LeapYears,&
+                    RecycStartYear, DtHrs, DirectRead
```

Comment:
Definition of namelist must be in definition section of file, i.e. move it to line 502.